Client/Frontend/UserContent/UserScripts/RequestBlocking.js

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+((args) => {
+  "use strict";
+  //[DEBUG]console.log('[RB] Loaded request blocking')
+  const securityToken = args.securityToken
+  const sendMessage = (resourceURL, resourceType) => {
+    //[DEBUG]console.log(`[RB] Checking if request should be blocked for '${resourceURL}'`)
+    return webkit.messageHandlers[args.handlerName].postMessage({
+      securityToken: securityToken,
+      resourceURL: resourceURL.href,
+      sourceURL: window.location.href,
+      resourceType: resourceType
+    })
+  }
+  
+  const makeURLFromString = (urlString) => {
+    try {
+      return new URL(urlString, window.location.href)
+    }
+    catch(err) {
+      //[DEBUG]console.log(`[RB] URL Error: '${err}'`)
+      return
+    }
+  }
+  
+  const { fetch: originalFetch } = window
+  window.fetch = function() {
+    const [resource, config] = arguments
+    
+    // Extract the url string
+    let urlString
+    if (typeof resource === 'string' || resource instanceof String) {
```

Comment:
Same question here, why all the type checking for all these values?  Why the uncertainty about data types?